test/CMakeLists.txt

```diff
@@ -0,0 +1,213 @@
+     test-shutdown-eof.c
+     test-shutdown-twice.c
+     test-signal-multiple-loops.c
+     test-signal-pending-on-close.c
+     test-signal.c
+     test-socket-buffer-size.c
+     test-spawn.c
+     test-stdio-over-pipes.c
+     test-strscpy.c
+     test-tcp-alloc-cb-fail.c
+     test-tcp-bind-error.c
+     test-tcp-bind6-error.c
+     test-tcp-close-accept.c
+     test-tcp-close-while-connecting.c
+     test-tcp-close.c
+     test-tcp-close-reset.c
+     test-tcp-connect-error-after-write.c
+     test-tcp-connect-error.c
+     test-tcp-connect-timeout.c
+     test-tcp-connect6-error.c
+     test-tcp-create-socket-early.c
+     test-tcp-flags.c
+     test-tcp-oob.c
+     test-tcp-open.c
+     test-tcp-read-stop.c
+     test-tcp-shutdown-after-write.c
+     test-tcp-try-write.c
+     test-tcp-try-write-error.c
+     test-tcp-unexpected-read.c
+     test-tcp-write-after-connect.c
+     test-tcp-write-fail.c
+     test-tcp-write-queue-order.c
+     test-tcp-write-to-half-open-connection.c
+     test-tcp-writealot.c
+     test-thread-equal.c
+     test-thread.c
+     test-threadpool-cancel.c
+     test-threadpool.c
+     test-timer-again.c
+     test-timer-from-check.c
+     test-timer.c
+     test-tmpdir.c
+     test-tty-duplicate-key.c
+     test-tty-escape-sequence-processing.c
+     test-tty.c
+     test-udp-alloc-cb-fail.c
+     test-udp-bind.c
+     test-udp-connect.c
+     test-udp-create-socket-early.c
+     test-udp-dgram-too-big.c
+     test-udp-ipv6.c
+     test-udp-multicast-interface.c
+     test-udp-multicast-interface6.c
+     test-udp-multicast-join.c
+     test-udp-multicast-join6.c
+     test-udp-multicast-ttl.c
+     test-udp-open.c
+     test-udp-options.c
+     test-udp-send-and-recv.c
+     test-udp-send-hang-loop.c
+     test-udp-send-immediate.c
+     test-udp-sendmmsg-error.c
+     test-udp-send-unreachable.c
+     test-udp-try-send.c
+     test-uname.c
+     test-walk-handles.c
+     test-watcher-cross-stop.c)
+
+add_executable(uv_run_tests ${uv_test_sources} uv_win_longpath.manifest)
+target_compile_definitions(uv_run_tests
+                           PRIVATE ${uv_defines} USING_UV_SHARED=1)
+target_compile_options(uv_run_tests PRIVATE ${uv_cflags})
+target_link_libraries(uv_run_tests uv ${uv_test_libraries})
+
+add_test(NAME uv_test
+         COMMAND uv_run_tests
+         WORKING_DIRECTORY ${PROJECT_SOURCE_DIR})
+if(CMAKE_SYSTEM_NAME STREQUAL "OS390")
+  set_tests_properties(uv_test PROPERTIES ENVIRONMENT
+                       "LIBPATH=${CMAKE_BINARY_DIR}:$ENV{LIBPATH}")
+endif()
+add_executable(uv_run_tests_a ${uv_test_sources} uv_win_longpath.manifest)
+target_compile_definitions(uv_run_tests_a PRIVATE ${uv_defines})
+target_compile_options(uv_run_tests_a PRIVATE ${uv_cflags})
+target_link_libraries(uv_run_tests_a PRIVATE uv_a ${uv_test_libraries} $<$<BOOL:${QEMU}>:-static>)
+add_test(NAME uv_test_a
+         COMMAND uv_run_tests_a
+         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
```

Comment:
```suggestion
         WORKING_DIRECTORY ${PROJECT_SOURCE_DIR})
```

Reply:
I'm mildly confused here. The diff already shows that the WORKING_DIRECTORY is set to `${PROJECT_SOURCE_DIR}` (you had suggested changing it to `${CMAKE_CURRENT_SOURCE_DIR}` above). Am I missing something or was some code committed that I'm unaware of?

Reply:
The same line of code appears twice (uv_test and uv_test_a). I'm assuming they needed the same change?

Reply:
oh you're totally right. I've clearly not had enough caffeine today 🤦‍♀️

I'll try to get to all of this + rebase this weekend. I'm swamped for this week workwise, sadly.

Reply:
apologies. Life and covid and other things have gotten in the way of me working on this. I'll hopefully return to this in the next week or two. Sorry!